google/cloud/forseti/scanner/scanners/blacklist_scanner.py

```diff
@@ -146,7 +152,11 @@ def _find_violations(self, instances_networks_data):
 
     def run(self):
         """Runs scanning."""
-        instances_network_interface_data = self._retrieve()
+        try:
```

Comment:
optional: I might be missing some background here, but is this behaviour consistent with other scanners? Most scanners just seem to call _retrieve and blindly pass in the result into the rule engine (even if it's empty), and the retrieve function does not check or raise any exceptions either. I wonder if we can do that? Calling the rule engine can still be desirable if there are rules with 'mode: required', etc.

If not, maybe just return the empty list from retrieve and make the check here
```
if not instances_network_data:
   LOGGER.info('No VM network interfaces found, nothing to do'.)
   return
```